src/validator/AuthTokenValidatorImpl.php

```diff
@@ -104,47 +102,42 @@ private function validateTokenLength(string $authToken): void
         }
     }
 
-    private function parseToken(string $authToken): WebEidAuthToken
-    {
-        try {
-            $token = new WebEidAuthToken($authToken);
-            if (is_null($token)) {
-                throw new AuthTokenParseException("Web eID authentication token is null");
-            }
-            return $token;
-        } catch (UnexpectedValueException $e) {
-            throw new AuthTokenParseException("Error parsing Web eID authentication token", $e);
-        }
-    }
-
     public function parse(string $authToken): WebEidAuthToken
     {
-        $this->logger->info("Starting token parsing");
+        if ($this->logger) {
+            $this->logger->info("Starting token parsing");
+        }
 
         try {
             $this->validateTokenLength($authToken);
-            return $this->parseToken($authToken);
+            return new WebEidAuthToken($authToken);
         } catch (Throwable $e) {
-            $this->logger->warning("Token parsing was interrupted: " . $e->getMessage());
+            if ($this->logger) {
+                $this->logger->warning("Token parsing was interrupted: " . $e->getMessage());
+            }
             throw $e;
         }
     }
 
     public function validate(WebEidAuthToken $authToken, string $currentChallengeNonce): X509
     {
-        $this->logger->info("Starting token validation");
+        if ($this->logger) {
+            $this->logger->info("Starting token validation");
+        }
```

Comment:
Let's use the [nullsafe operator](https://wiki.php.net/rfc/nullsafe_operator) everywhere:
```suggestion
        $this->logger?->info("Starting token validation");
```
Can you please to a global search/replace across the whole project?